server/SmartHome.Data/SmartHomeContext.cs

```diff
@@ -0,0 +1,244 @@
+
+        }
+
+        private void OnModuleCreated(ModelBuilder modelBuilder)
+        {
+            var entity = modelBuilder.Entity<Module>();
+            entity
+               .ToTable("Module");
+            entity
+                .HasKey(x => x.ModuleId);
+
+            entity
+                .Property(x => x.IsConnected)
+                .IsRequired();
+
+            entity
+                .HasOne<Room>(x => x.Room)
+                .WithMany(x => x.Modules)
+                .HasForeignKey(x => x.RoomId)
+                .IsRequired();
+
+        }
+        private void OnRoomCreated(ModelBuilder modelBuilder)
+        {
+            var entity = modelBuilder.Entity<Room>();
+            entity
+               .ToTable("Room");
+            entity
+                .HasKey(x => x.RoomId);
+
+            entity
+                .Property(x => x.Name)
+                .HasMaxLength(30)
+                .IsRequired();
+
+            entity
+                .HasOne<SmartHomeEntity>(x => x.SmartHomeEntity)
+                .WithMany(x => x.Rooms)
+                .HasForeignKey(x => x.SmartHomeEntityId)
+                .IsRequired();
+        }
+
+        private void OnSmartHomeEntityCreated(ModelBuilder modelBuilder)
+        {
+            var entity = modelBuilder.Entity<SmartHomeEntity>();
+            entity
+               .ToTable("SmartHomeEntity");
+            entity
+                .HasKey(x => x.SmartHomeEntityId);
+
+            entity
+                .Property(x => x.RegisterTimestamp)
+                .IsRequired();
+
+            entity
+                .Property(x => x.IpAddress)
+                .HasMaxLength(15)
+                .IsFixedLength(true)
+                .IsRequired();
+           
+        }
+
+        private void OnUserCreated(ModelBuilder modelBuilder)
+        {
+            var entity = modelBuilder.Entity<User>();
+            entity
+               .ToTable("User");
+            entity
+                .HasKey(x => x.UserId);
+
+            entity
+               .Property(x => x.Username)
+               .HasMaxLength(20)
+               .IsRequired();
+            entity
+                .HasIndex(x => x.Username)
+                .IsUnique();
+            entity
+                .HasIndex(x => x.Email)
+                .IsUnique();
+
+
+
+            entity
+                .Property(x => x.Password)
+                .HasMaxLength(64)
+                .IsFixedLength(true)
+                .IsRequired();
+
+            entity
+                .Property(x => x.Email)
+                .HasMaxLength(40)
+                .IsRequired();
+
+            entity
+                .Property(x => x.IsAdmin)
+                .IsRequired();
+        }
+        private void OnUserSmartHomeEntityCreated(ModelBuilder modelBuilder)
+        {
+            
+            var entity = modelBuilder.Entity<UserSmartHomeEntity>();
+            entity
+               .ToTable("UserSmartHomeEntity");
```

Comment:
Remove

Reply:
nope, without this EF uses plural for table names